corehq/apps/reports/dispatcher.py

```diff
@@ -87,8 +87,8 @@ def get_reports(self, domain=None):
             project = None
 
         def process(reports):
-            if project and callable(reports):
-                reports = reports(project)
+            if callable(reports):
+                reports = reports(project) if project else tuple()
```

Comment:
Today I learned that `tuple(tuple())` (which gets returned if project is falsy) is equivalent to `tuple()`. Interesting.

Reply:
haha, yeah, `tuple()` is a constructor , so it's roughly equivalent to a tuple literal `()`.  I just figured `tuple()` was more intuitive than `()` since you never really see the latter for some reason.  The second invocation (line 92) performs type coercion if `reports` isn't already a tuple.

This whole business about not knowing whether reports is even a callable or not is really code-smelly, but I didn't wanna go too deep and start refactoring all this.